runtime/doc/builtin.txt

```diff
@@ -4261,6 +4263,36 @@ getreginfo([{regname}])					*getreginfo()*
 		Can also be used as a |method|: >
 			GetRegname()->getreginfo()
 
+getregion({pos1}, {pos2}, {type})			*getregion()*
+		Returns the list of strings from {pos1} to {pos2}.
+		For possible values of {pos1} and {pos2} see |line()|.
+		The {type} is region type:
+			"v" for |charwise| mode
+			"V" for |linewise| mode
+			"<CTRL-V>" for |blockwise-visual| mode
```

Comment:
It is not clear to me how the specified visual mode type is used.  I can see that irrespective of the current visual mode and the mode type argument, getregion() always returns the visually selected region of text.  I expected that if a region of text is selected using character-wise visual mode and getregion() is called with "V", then it will return an empty List.

Reply:
`{pos1}`, `{pos2}` is just used as position.
position `v` means `use visual mode start position`.

Reply:
I am referring to the "v" or "V" value in the new {type} argument.  For example, you select some characters using character-wise visual mode and then call getregion() with the "V" or "<CTR-V>" for the {type} argument, what value will it return?

Reply:
It depends on your cursor position.

Reply:
In my previous suggestion (https://github.com/vim/vim/pull/13998#issuecomment-1941481650), the {type} argument is for specifying which style is used to get text from {pos1} to {pos2}.  In other words, if you pass "V" for the {type} argument getregion() will return text from {pos1} to {pos2} as if it's selected in line-wise visual mode regardless of the current visual mode.  (So `getregion({pos1}, {pos2}, "V")` is also equal to `getline(getpos({pos1})[1], getpos({pos2})[1])`.)

At least I tried, the current implementation of getregion() follows this, so I think getregion() call with "V" for the {type} argument should return text of lines between the line of {pos1} and the line of {pos2}.

About getregion() call with "" for the {type} argument, I don't have an absolute one answer but I think the behavior should be one of these:
- Give an error like "{type} argument cannot be empty" or "Invalid value for the {type} argument"
- Use the current visual mode in visual mode and chararacter-wise mode in the other mode
- Use the current visual mode in visual mode and the last visual mode in the other mode

Reply:
> In my previous suggestion ([#13998 (comment)](https://github.com/vim/vim/pull/13998#issuecomment-1941481650)), the {type} argument is for specifying which style is used to get text from {pos1} to {pos2}. In other words, if you pass "V" for the {type} argument getregion() will return text from {pos1} to {pos2} as if it's selected in line-wise visual mode regardless of the current visual mode. (So `getregion({pos1}, {pos2}, "V")` is also equal to `getline(getpos({pos1})[1], getpos({pos2})[1])`.)
> 
> At least I tried, the current implementation of getregion() follows this, so I think getregion() call with "V" for the {type} argument should return text of lines between the line of {pos1} and the line of {pos2}.
> 

A few tests should be added for this.  Make a characterwise selection, call getregion() with the "v", "V" and "<C-V>" types and check the results.  Do the same for the linewise and blockwise selections.

Reply:
The help text also needs to be updated to clarify the use of the "type" argument.  Irrespective of the current visual mode, the supplied "type" will be used to determine the region of text returned.

Reply:
> A few tests should be added for this. Make a characterwise selection, call getregion() with the "v", "V" and "" types and check the results. Do the same for the linewise and blockwise selections.

I have added this.

Reply:
> The help text also needs to be updated to clarify the use of the "type" argument. Irrespective of the current visual mode, the supplied "type" will be used to determine the region of text returned.

I have updated the documentation.  But the suggestion is wellcome.